bindings/generate_bindings.py

```diff
@@ -301,7 +351,9 @@ def gen_def(self) -> Code:
         for name in required + optional:
             out += ["    " + self.params[name].gen_function_param()]
         out += ["    ):"]
-        out += [f"        self.{k} = {k}" for k in self.params]
+        for name in required + optional:
+            p = self.params[name]
+            out += [f"        self.{name} = {p.type.input_to_output(name, not p.required)}"]
```

Comment:
Note that before, the order of class `__init__` parameters did not match the order in which parameters were saved to the `self` object, which was annoying when inspecting the generated code.

I changed it, but it also caused a lot more churn in the generated bindings, sorry.

Reply:
this is now the second two commits, to make reviewing easier.